istioctl/cmd/tag.go

```diff
@@ -0,0 +1,400 @@
+			}
+			if len(args) > 1 {
+				return fmt.Errorf("can only provide a single tag for creation")
+			}
+
+			client, err := kubeClient(kubeconfig, configContext)
+			if err != nil {
+				return fmt.Errorf("failed to create kubernetes client: %v", err)
+			}
+
+			return setTag(context.Background(), client.Kube(), args[0], revision, cmd.OutOrStdout())
+		},
+	}
+
+	cmd.PersistentFlags().BoolVar(&overwrite, "overwrite", false, "whether to overwrite an existing tag")
+	cmd.PersistentFlags().BoolVarP(&skipConfirmation, "skip-confirmation", "y", false, "whether to skip confirmation for tag deletion")
+	cmd.PersistentFlags().StringVarP(&revision, "revision", "r", "", "revision to point tag to")
+	_ = cmd.MarkPersistentFlagRequired("revision")
+
+	return cmd
+}
+
+func tagListCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "List existing revision tags and their corresponding revisions",
+		Example: "istioctl x tag list",
+		Aliases: []string{"show"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 0 {
+				return fmt.Errorf("tag list command does not accept arguments")
+			}
+
+			client, err := kubeClient(kubeconfig, configContext)
+			if err != nil {
+				return fmt.Errorf("failed to create kubernetes client: %v", err)
+			}
+			return listTags(context.Background(), client.Kube(), cmd.OutOrStdout())
+		},
+	}
+
+	return cmd
+}
+
+func tagRemoveCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "remove",
+		Short:   "Remove an existing revision tag",
+		Example: "istioctl x tag remove prod",
+		Aliases: []string{"delete"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return fmt.Errorf("must provide a tag for removal")
+			}
+			if len(args) > 1 {
+				return fmt.Errorf("must provide a single tag for removal")
+			}
+
+			client, err := kubeClient(kubeconfig, configContext)
+			if err != nil {
+				return fmt.Errorf("failed to create kubernetes client: %v", err)
+			}
+
+			return removeTag(context.Background(), client.Kube(), args[0], skipConfirmation, cmd.OutOrStdout())
+		},
+	}
+
+	return cmd
+}
+
+// setTag creates or modifies a revision tag.
+func setTag(ctx context.Context, kubeClient kubernetes.Interface, tag, revision string, w io.Writer) error {
+	revWebhooks, err := getWebhooksWithRevision(ctx, kubeClient, revision)
+	if err != nil {
+		return err
+	}
+	if len(revWebhooks) == 0 {
+		return fmt.Errorf("cannot find webhook with revision %q", revision)
+	}
+	if len(revWebhooks) > 1 {
+		return fmt.Errorf("found multiple canonical webhooks for revision %q", revision)
+	}
+
+	tagWebhook, err := buildTagWebhookFromCanonical(revWebhooks[0], tag, revision)
+	if err != nil {
+		return err
+	}
+
+	whs, err := getWebhooksWithTag(ctx, kubeClient, tag)
+	if err != nil {
+		return err
+	}
+
+	if len(whs) > 0 && !overwrite {
+		return fmt.Errorf("found an existing revision tag for %q, use the --overwrite flag to overwrite", tag)
```

Comment:
Nit: The error from `kubectl label` when this happens is "error: 'foo' already has a value (bar), and --overwrite is false" which is somewhat more clear.